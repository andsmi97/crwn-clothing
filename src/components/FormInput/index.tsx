import React from "react";

import "./styles.scss";
import { IFormInput } from "../../types";

export default ({ onChange, label, ...restProps }: IFormInput): JSX.Element => {
  return (
    <div className="group">
      <input
        className="form-input"
        onChange={event => onChange && onChange(event.target.value)}
        {...restProps}
      />
      {label ? (
        <label
          htmlFor={label}
          className={`${
            restProps.value ? (restProps.value.length ? "shrink" : "") : ""
          } form-input-label`}
        >
          {label}
        </label>
      ) : null}
    </div>
  );
};
