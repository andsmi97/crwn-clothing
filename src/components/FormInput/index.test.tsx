import React from "react";
import { shallow, ShallowWrapper } from "enzyme";
import FormInput from "./index";

describe("<FormInput />", () => {
  const wrapper = shallow(<FormInput />);
  it("matches snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("renders label given label", () => {
    const wrapper = shallow(<FormInput label="label" />);
    console.log(wrapper.find("label"));
    expect(wrapper.find("label")).toBe(1);
  });
});
