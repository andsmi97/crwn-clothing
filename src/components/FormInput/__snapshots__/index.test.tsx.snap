// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormInput /> matches snapshot 1`] = `ShallowWrapper {}`;
